yastd/dbus_api.md

```diff
@@ -0,0 +1,169 @@
+
+## get the list of available languages
+
+Iface: o.o.YaST.Installer1.Language
+
+methods:
+
+  ToInstall(array(string)) -> void
+    Set list of languages to install
+    Example:
+      ToInstall(["cs_CZ", "de_DE"]) -> () # only lang codes from AvailableLanguages is supported
+
+Properties (all read only):
+
+  AvailableLanguages -> array(array(string, string, dict(string, variant)))
+    List of all available languages to install on target system.
+    Example:
+      AvailableLanguages -> [["cs_CZ", "Czech", {}]] # it is lang code, human readable lang name and dict for future extensions to provide more data
+
+  MarkedForInstall -> array(string)
+    List of languages to install. Same format as ToInstall
+
+Signals:
+
+  PropertiesChanged ( only standard one from org.freedesktop.DBus.Properties interface )
+
+
+notes:
+
+identifiers: maybe LanguageTag https://www.rubydoc.info/github/yast/yast-packager/master/LanguageTag
+- move it to yast-yast2
+- link to the standard from yard
+- see https://tools.ietf.org/html/rfc4647 Matching of Language Tags
+- see https://lists.opensuse.org/archives/list/yast-devel@lists.opensuse.org/message/D52PSZ7TRID2RVM6CE6K2C2RUNNGOS6Z/
+
+## get the list of base products
+
+Iface: o.o.YaST.Installer1.Software
+
+methods:
+
+  InstallProduct(string) -> void
+    Set product for install.
+    TODO: do we need version or arch or any other parameter? should we add generic dict(sv) to it?
+    Example:
+      InstallProduct("SLES") -> () # only name from available BaseProducts is supported
+
+Properties (all read only):
+
+  AvailableBaseProducts -> array(array(string, string, dict(string, variant)))
+    List of all available base product to install on target system.
+    Note: List is sorted according to defined display order
+    Example:
+      AvailableBaseProducts -> [["SLES", "SUSE Linux Enterprise Server", {}]] # it is product name, human readable name and dict for future extensions to provide more data
```

Comment:
Again, a struct like for languages.
How about this naming:
- ProductKey or ProductIdentifier
- product_label (where **label** is a "string visible by the user"; I try to avoid **name** because it can mean "label" or "key")
- details